diff --git a/bin/democratic-csi b/bin/democratic-csi
index c1c23b1..6b5a67b 100755
--- a/bin/democratic-csi
+++ b/bin/democratic-csi
@@ -9,6 +9,7 @@
 require("../src/utils/polyfills");
 const yaml = require("js-yaml");
 const fs = require("fs");
+const { substituteEnvVars } = require("../src/utils/config_env");
 const { grpc } = require("../src/utils/grpc");
 const { stringify, stripWindowsDriveLetter } = require("../src/utils/general");

@@ -32,7 +33,7 @@ const args = require("yargs")
       }

       try {
-        options = yaml.load(fs.readFileSync(path, "utf8"));
+        options = yaml.load(substituteEnvVars(fs.readFileSync(path, "utf8")));
         try {
           driverConfigFile = fs.realpathSync(path);
         } catch (e) {
diff --git a/src/utils/config_env.js b/src/utils/config_env.js
new file mode 100644
index 0000000..b1406e1
--- /dev/null
+++ b/src/utils/config_env.js
@@ -0,0 +1,11 @@
+function substituteEnvVars(config) {
+  return config.replace(/{env:(.*)}/gm, (match, varName) => {
+    if (!(varName in process.env)) {
+      return match;
+    }
+
+    return process.env[varName];
+  });
+}
+
+module.exports.substituteEnvVars = substituteEnvVars;
